### DISCUSSION TITLE
firefox rejected v0.4.5

### DISCUSSION THREAD

**Discussion by kheina** [kheina-com/Blue-Blocker#327] [Announcements]
Hello,

Your Extension Blue Blocker was manually reviewed by the Mozilla Add-ons team in an assessment performed on our own initiative of content that was submitted to Mozilla Add-ons.

Our review found that your content violates the following Mozilla policy or policies:

- Sources, specifically Build not reproducing: Reproducing the submitted release version based on the provided source code package and instructions failed.

You can access the console output at https://paste.mozilla.org/cOG4ewPP
Environment used for building: Node 18.20.4,  npm 10.7.0 on Ubuntu 22.04 LTS(10GB RAM, 6 CPUs)

Please test your build in a clean environment to make sure it is reproducible. If necessary, update the source code package and/or the instructions to reproduce.
Please read through the instructions at https://extensionworkshop.com/documentation/publish/source-code-submission/ ..

Affected versions: 0.4.5

Based on that finding, those versions of your Extension will be disabled on https://addons.mozilla.org/addon/blue-blocker/ in 14 day(s). Once disabled, these versions will no longer be available for download from Mozilla Add-ons, anywhere in the world, but any compliant versions will remain in place, and you may choose to upload a new version that addresses the policy violation. Users who have previously installed the disabled versions will be able to continue using them.

More information about Mozilla's add-on policies can be found at https://extensionworkshop.com/documentation/publish/add-on-policies/.

Thank you for your attention.

[ref:55f1d205-a149-4b4c-8f0b-7ed47e9ef5ea]
--
Mozilla Add-ons Team
https://addons.mozilla.org/

**Comment by kheina** [comment]
actually, I don't think this is a full rejection, but if someone could take a look that would be great @cooljeanius @rougetimelord

**Reply by rougetimelord** [reply]
Weird, might be because we switched to html-minifier-terser?? That should mean just shooting off a new release now

**Reply by cooljeanius** [reply]
This line looks suspicious:
```
/bin/sh: 1: jq: not found
```
Where would it be best to explicitly declare a dependency on `jq`? AFAIK that's more of a system package manager thing than it is an npm thing? 
Edit: oh wait no, I see they install it via npm next...

**Reply by cooljeanius** [reply]
I see they're also getting an error from `make`; @kheina could you take a look at #308 please?

**Reply by cooljeanius** [reply]
I think there's a legitimate build failure from `tsc` here; I get the same failure on both my system and on the CI workflow I'm working on in #328...

**Reply by isaac-mok** [reply]
> This line looks suspicious:
> 
> ```
> /bin/sh: 1: jq: not found
> ```
> 
> Where would it be best to explicitly declare a dependency on `jq`? AFAIK that's more of a system package manager thing than it is an npm thing? Edit: oh wait no, I see they install it via npm next...

`jq` on npm is a completely different thing. They need to install it via `apt-get`.

**Reply by cooljeanius** [reply]
OK, I opened #333 to document that in the README

**Reply by topokel** [reply]
> I think there's a legitimate build failure from `tsc` here; I get the same failure on both my system and on the CI workflow I'm working on in #328...

All of the subsequent errors actually appear to come from the erroneously installed jquery library.

There's [jq-wasm](https://www.npmjs.com/package/jq-wasm) which might be a possible replacement for expecting the user to install the actual `jq` binary themselves, although it seems quite a silly change maybe it could help with Firefox addon submission at the same time.

EDIT: after looking at the npm page for jq-wasm for two seconds it is only used by like 5 people so.... user beware lol. Like I said, would be somewhat silly of a change just to get firefox submission through.

**Comment by kheina** [comment]
looks like other people are having issues too....

https://www.neowin.net/amp/ublock-origin-lite-maker-ends-firefox-store-support-slams-mozilla-for-hostile-reviews/

**Reply by cooljeanius** [reply]
oldtwitter, too: https://bsky.app/profile/cybershot.bsky.social/post/3l5iagr2icn22

**Reply by rougetimelord** [reply]
hey! maybe firefox AMO will finally stop being impossible to please

**Comment by rougetimelord** [comment]
Hey @kheina wanted to follow up, is there any new stuff from Mozilla?

**Reply by rougetimelord** [reply]
PS: Sending you an email about the potential of adding me to the addon ownership so you don't have to relay messages uwu. Different email than last time, .cool TLD
